packages/web/src/alliance/Members.tsx

```diff
@@ -0,0 +1,238 @@
+import { TFunction } from 'next-i18next'
+import * as React from 'react'
+import LazyFade from 'react-lazyload-fadein'
+import { Image, StyleSheet, Text, View } from 'react-native'
+import Ally from 'src/alliance/AllianceMember'
+import { Category as CategoryEnum } from 'src/alliance/CategoryEnum'
+import { H2, H4 } from 'src/fonts/Fonts'
+import { NameSpaces, Trans, useTranslation } from 'src/i18n'
+import { Cell, GridRow, Spans } from 'src/layout/GridRow'
+import { useScreenSize } from 'src/layout/ScreenSize'
+import { ListItem } from 'src/shared/DropDown'
+import DropDownGroup from 'src/shared/DropDownGroup'
+import Outbound, { externalizeURL } from 'src/shared/Outbound'
+import { colors, fonts, standardStyles, textStyles } from 'src/styles'
+
+async function gatherAllies(persistFunc: (data: []) => void, signal: { aborted: boolean }) {
+  const response = await fetch('api/alliance')
+  const alliesByCategory = await response.json()
+  if (!signal.aborted) {
+    persistFunc(alliesByCategory)
+  }
+}
+
+function buildDropDownProps(t: TFunction, currentFilter: string): ListItem[] {
+  return Object.keys(CategoryEnum).map((key) => {
+    return {
+      id: key,
+      selected: key === currentFilter,
+      label: t(`members.categoryTitle.${key.toLocaleLowerCase()}`),
+    }
+  })
+}
+
+function initialState() {
+  return Object.keys(CategoryEnum).map((key: CategoryEnum) => {
+    return {
+      name: key,
+      records: [],
+    }
+  })
+}
+
+const ALL = 'all'
+
+export default function Members() {
+  const { t } = useTranslation(NameSpaces.alliance)
+  const [alliesByCategory, setAllies] = React.useState(initialState())
+  const [selectedFilter, setFilter] = React.useState(ALL)
+
+  const { isTablet, isMobile } = useScreenSize()
+
+  React.useEffect(() => {
+    const signal = { aborted: false }
+    // sometimes it is nessessary to break a rule
+    // https://github.com/facebook/react/issues/14326
+    // tslint:disable-next-line: no-floating-promises
+    gatherAllies(setAllies, signal)
```

Comment:
I had a look at the react link you linked to. This doesn't match the async effect pattern Abramov provided. Why let this promise float?

Reply:
here is my understanding.  `useEffect` is typed to take a function that returns void or a cleanup function. NOT a promise. so cant do await on this as that would require an async function which always return promises.